AdminServer/appscale/admin/utils.py

```diff
@@ -401,3 +403,83 @@ def assign_ports(old_version, new_version, zk_client):
 
   return {'httpPort': new_http_port, 'httpsPort': new_https_port,
           'haproxyPort': haproxy_port}
+
+
+def validate_queue(queue):
+  """ Checks if a queue configuration is valid.
+
+  Args:
+    queue: A dictionary containing queue configuration details.
+  Raises:
+    InvalidQueueConfiguration if configuration is invalid.
+  """
+  mode = queue.get('mode', 'push')
+
+  if mode not in ['push', 'pull']:
+    raise InvalidQueueConfiguration('Invalid queue mode: {}'.format(mode))
+
+  if mode == 'push':
+    required_fields = tq_constants.REQUIRED_PUSH_QUEUE_FIELDS
+    supported_fields = tq_constants.SUPPORTED_PUSH_QUEUE_FIELDS
+  else:
+    required_fields = tq_constants.REQUIRED_PULL_QUEUE_FIELDS
+    supported_fields = tq_constants.SUPPORTED_PULL_QUEUE_FIELDS
+
+  for field in required_fields:
+    if field not in queue:
+      raise InvalidQueueConfiguration(
+        'Queue is missing {}: {}'.format(field, queue))
+
+  for field in queue:
```

Comment:
This package is tagged as being Python 3 compatible, and iteritems was renamed in Python 3. I could use the six module, but I don't think the additional dependency is worth it yet.